plugin/pkg/shoot/dns/admission.go

```diff
@@ -245,6 +252,24 @@ func checkFunctionlessDNSProviders(dns *core.DNS) error {
 	return nil
 }
 
+func checkIfShootMigrationIsPossible(seedLister corelisters.SeedLister, oldShoot, newShoot *core.Shoot) error {
+	for _, seedName := range []string{*oldShoot.Spec.SeedName, *newShoot.Spec.SeedName} {
+		seedDNSDisabled, err := seedDisablesDNS(seedLister, seedName)
+		if err != nil {
+			return apierrors.NewBadRequest(fmt.Sprintf("could not get referenced seed: %+v", err.Error()))
+		}
+		if seedDNSDisabled {
+			return apierrors.NewBadRequest("source and destination seeds must enable DNS so that the shoot can be migrated")
+		}
+	}
+
+	if newShoot.Spec.DNS != nil {
+		return checkFunctionlessDNSProviders(newShoot.Spec.DNS)
+	}
```

Comment:
I actually would like to refrain from calling this here because the admission code is already pretty nested, and responsibility wise `checkIfShootMigrationIsPossible` shouldn't call any provider validation.

Reply:
Just so you know why I called it here: 
Generally, if the seed is updated nothing prevents the user to also update the DNS Providers.
So when the seed is updated I wanted to check if migration is possible, check if the providers are ok and then always return from the function so that we don't continue and generate a default domain if the DNS of the shoot happens to be nil.

Reply:
Yes, I'm not objecting the use case but I think you can also check the DNS providers first, no matter if shoot migration is initiated or not and call `checkIfShootMigrationIsPossible` if necessary.